packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -22,7 +22,8 @@ export interface AwsS3STSResponse {
   region: string
 }
 
-export interface AwsS3MultipartOptions extends PluginOptions {
```

Comment:
`PluginOptions` contains only an `id` and a `replaceTargetContent` fields. Given that `replaceTargetContent` is not used / documented I removed the inheritance.

Reply:
I think we want to keep the inheritence, even if we don't use the parents definitions, it's still useful to match the type `PluginOptions`.

Reply:
@Murderlon is considering removing `UploaderPlugin` in https://github.com/transloadit/uppy/pull/4573, so maybe we shouldn't include it in our plans

Reply:
Ok - i pushed new types such that classes that extend `UIPlugin` take `UIPluginOptions`; classes that extend `BasePlugin` take `PluginOptions`. Other plugins have been updated with this type information though I did not check for type correctness beyond updating plugins that cited inheriting from `UIPlugin` to have their options inherit from `UIPluginOptions`

Reply:
There is one exception to that `uppy/packages/@uppy/google-drive/types/index.test-d.ts` should probably be updated as it defines `SomePlugin extends<PluginOptions>`   for testing purposes. However if i touched that file (including just adding a comment) I got a lint error 

```
/home/bdirito/dev/locution/uppy/packages/@uppy/google-drive/types/index.test-d.ts
  1:47  error  Unable to resolve path to module '@uppy/core'  import/no-unresolved
  2:25  error  Unable to resolve path to module '..'          import/no-unresolved
```

This is strange as the pr already alters `uppy/packages/@uppy/aws-s3/types/index.test-d.ts`. I did not see any differences in their `packages.json` that might account for this but I dont have a ton of time at the moment.

Reply:
It looks like running `build:lib` as in the github ci action exports things correctly to allow for this edit. Fixed.